AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -209,24 +182,48 @@ public class AWSAppSyncClientConfiguration {
                 }
                 defaultApiKeyAuthProvider = BasicAWSAPIKeyAuthProvider(key: appSyncClientInfo.apiKey)
             }
-        }
-        
-        // If AuthType is AWS_IAM, use the AWSCredentialsProvider passed in
-        // or create a provider based on the CognitoIdentity CredentialsProvider
-        // passed from the config
-        var defaultCredentialsProvider: AWSCredentialsProvider? = credentialsProvider
-        if authTypeFromConfig == AuthType.awsIAM {
+            break
+        case AuthType.amazonCognitoUserPools:
+            if credentialsProvider != nil || apiKeyAuthProvider != nil || oidcAuthProvider != nil {
+                throw AWSAppSyncClientInfoError(errorMessage: AuthType.amazonCognitoUserPools.rawValue + " is selected in configuration but a "
+                    + "AWSCognitoUserPoolsAuthProvider object is not passed in.")
```

Comment:
Should we change the message to indicate that an un-supported auth provider object is passed in?

Reply:
I want to be a little bit more explanatory in the error. Hence explaining it.